# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/libwebrtcglue/WebrtcMediaCodecVP8VideoCodec.h
# Commit: 9d8f6e354d8c
# Full Hash: 9d8f6e354d8c332bab7e433252077ec0e1fa8f04
# Author: John Lin <jolin@mozilla.com>
# Date: 2021-11-01 18:17:33
# Regressor Bug: 1654112
# File Overlap Count: 1
# Description:
#   Bug 1654112 - remove MediaCodecVP8Codec. r=mjf
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D115531
# ==============================================================================

diff -r 7675ead797c8 -r 9d8f6e354d8c dom/media/webrtc/libwebrtcglue/WebrtcMediaCodecVP8VideoCodec.h
--- a/dom/media/webrtc/libwebrtcglue/WebrtcMediaCodecVP8VideoCodec.h	Fri Oct 08 15:57:53 2021 +0200
+++ /dev/null	Thu Jan 01 00:00:00 1970 +0000
@@ -1,137 +0,0 @@
-/* This Source Code Form is subject to the terms of the Mozilla Public
- * License, v. 2.0. If a copy of the MPL was not distributed with this file,
- * You can obtain one at http://mozilla.org/MPL/2.0/. */
-
-#ifndef WebrtcMediaCodecVP8VideoCodec_h__
-#define WebrtcMediaCodecVP8VideoCodec_h__
-
-#include <jni.h>
-
-#include "mozilla/java/CodecProxyWrappers.h"
-#include "mozilla/Mutex.h"
-#include "nsThreadUtils.h"
-
-#include "MediaConduitInterface.h"
-#include "AudioConduit.h"
-#include "VideoConduit.h"
-
-#include "modules/video_coding/include/video_codec_interface.h"
-
-namespace mozilla {
-
-struct EncodedFrame {
-  uint32_t width_;
-  uint32_t height_;
-  uint32_t timeStamp_;
-  uint64_t decode_timestamp_;
-};
-
-class WebrtcAndroidMediaCodec;
-
-class WebrtcMediaCodecVP8VideoEncoder : public WebrtcVideoEncoder {
- public:
-  WebrtcMediaCodecVP8VideoEncoder();
-
-  virtual ~WebrtcMediaCodecVP8VideoEncoder() override;
-
-  // Implement VideoEncoder interface.
-  virtual int32_t InitEncode(
-      const webrtc::VideoCodec* codecSettings,
-      const webrtc::VideoEncoder::Settings& settings) override;
-
-  virtual int32_t Encode(
-      const webrtc::VideoFrame& inputImage,
-      const std::vector<webrtc::VideoFrameType>* frame_types) override;
-
-  virtual int32_t RegisterEncodeCompleteCallback(
-      webrtc::EncodedImageCallback* callback) override;
-
-  virtual int32_t Release() override;
-
-  virtual void SetRates(
-      const webrtc::VideoEncoder::RateControlParameters& aParameters) override;
-
- private:
-  int32_t VerifyAndAllocate(const uint32_t minimumSize);
-  bool ResetInputBuffers();
-  bool ResetOutputBuffers();
-
-  size_t mMaxPayloadSize;
-  webrtc::EncodedImage mEncodedImage;
-  webrtc::EncodedImageCallback* mCallback;
-  uint32_t mFrameWidth;
-  uint32_t mFrameHeight;
-
-  WebrtcAndroidMediaCodec* mMediaCodecEncoder;
-
-  jobjectArray mInputBuffers;
-  jobjectArray mOutputBuffers;
-};
-
-class WebrtcMediaCodecVP8VideoRemoteEncoder : public WebrtcVideoEncoder {
- public:
-  WebrtcMediaCodecVP8VideoRemoteEncoder()
-      : mConvertBuf(nullptr), mConvertBufsize(0), mCallback(nullptr) {}
-
-  ~WebrtcMediaCodecVP8VideoRemoteEncoder() override;
-
-  // Implement VideoEncoder interface.
-  int32_t InitEncode(const webrtc::VideoCodec* codecSettings,
-                     const webrtc::VideoEncoder::Settings& settings) override;
-
-  int32_t Encode(const webrtc::VideoFrame& inputImage,
-                 const std::vector<webrtc::VideoFrameType>* frame_types) override;
-
-  int32_t RegisterEncodeCompleteCallback(
-      webrtc::EncodedImageCallback* callback) override;
-
-  int32_t Release() override;
-
-  void SetRates(
-      const webrtc::VideoEncoder::RateControlParameters& aParameters) override;
-
- private:
-  java::CodecProxy::GlobalRef mJavaEncoder;
-  java::CodecProxy::NativeCallbacks::GlobalRef mJavaCallbacks;
-  uint8_t* mConvertBuf;
-  uint8_t mConvertBufsize;
-  webrtc::EncodedImageCallback* mCallback;
-};
-
-class WebrtcMediaCodecVP8VideoDecoder : public WebrtcVideoDecoder {
- public:
-  WebrtcMediaCodecVP8VideoDecoder();
-
-  virtual ~WebrtcMediaCodecVP8VideoDecoder() override;
-
-  virtual int32_t InitDecode(const webrtc::VideoCodec* codecSettings,
-                             int32_t numberOfCores) override;
-
-  virtual int32_t Decode(
-      const webrtc::EncodedImage& inputImage, bool missingFrames,
-      int64_t renderTimeMs = -1) override;
-
-  virtual int32_t RegisterDecodeCompleteCallback(
-      webrtc::DecodedImageCallback* callback) override;
-
-  virtual int32_t Release() override;
-
- private:
-  void DecodeFrame(EncodedFrame* frame);
-  void RunCallback();
-  bool ResetInputBuffers();
-  bool ResetOutputBuffers();
-
-  webrtc::DecodedImageCallback* mCallback;
-
-  uint32_t mFrameWidth;
-  uint32_t mFrameHeight;
-
-  WebrtcAndroidMediaCodec* mMediaCodecDecoder;
-  jobjectArray mInputBuffers;
-  jobjectArray mOutputBuffers;
-};
-
-}  // namespace mozilla
-
-#endif  // WebrtcMediaCodecVP8VideoCodec_h__